torch/onnx/_internal/fx/decomposition_skip.py

```diff
@@ -0,0 +1,139 @@
+"""A context manager that disables the decomposition of certain ops during dynamo tracing.
+
+The approach is to temporarily hijack the operator callable with PT2 custom operator.
+The custom operator will not be decomposed and will show up as a single node to be exported to ONNX.
+
+For the time being the decomposition of these ops is otherwise unavoidable.
+
+https://github.com/pytorch/pytorch/issues/116684
+https://github.com/pytorch/pytorch/issues/115883
+
+This solution will no longer be required once the issue is resolved.
+"""
+from __future__ import annotations
+
+import abc
+import contextlib
+
+from typing import Callable, Sequence, Type
+
+from onnxscript.function_libs.torch_lib.ops import (  # type: ignore[import-not-found]
+    nn as torchlib_nn,
+)
+
+import torch
+from torch._decomp import decompositions
+
+_NEW_OP_NAMESPACE: str = "onnx_export"
+"""The namespace for the custom operator."""
+
+
+class DecompSkip:
```

Comment:
```suggestion
class DecompSkip(abc.ABC):
```

was this abc?

Reply:
I'll likely add this in a follow up now that all tests are green.. Depending on if we have permission to merge.